service-front/module/Application/src/Controller/VouchingFlowController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Controller;
+
+use Application\Contracts\OpgApiServiceInterface;
+use Application\Controller\Trait\FormBuilder;
+use Application\Forms\ConfirmVouching;
+use Laminas\Http\Response;
+use Laminas\Mvc\Controller\AbstractActionController;
+use Laminas\View\Model\ViewModel;
+
+class VouchingFlowController extends AbstractActionController
+{
+    use FormBuilder;
+
+    protected $plugins;
+
+    public function __construct(
+        private readonly OpgApiServiceInterface $opgApiService,
+    ) {
+    }
+
+    public function confirmVouchingAction(): ViewModel|Response
+    {
+        $view = new ViewModel();
+        $uuid = $this->params()->fromRoute("uuid");
+        $details_data = $this->opgApiService->getDetailsData($uuid);
+        $form = $this->createForm(ConfirmVouching::class);
+
+        $view->setVariable('details_data', $details_data);
+        /**
+         * @psalm-suppress InvalidArrayOffset
+         */
+        $view->setVariable('vouching_for', $details_data["vouchingFor"]);
+        $view->setVariable('form', $form);
+
+        if ($this->getRequest()->isPost()) {
+            $formData = $this->getRequest()->getPost();
+
+            if (isset($formData['try_different'])) {
+                return $this->redirect()->toUrl(
```

Comment:
should i be using `toRoute` instead of `toUrl` here - I did give it a go but couldn't make a success of it.

Reply:
So this doesn't work?
```
return $this->redirect()->toRoute("start", [
    'personType' => 'donor',
    'lpas' => $details_data['lpas']
]);
```

Reply:
nope - and i couldn't figure out why,

Reply:
Looking at this further... Wouldn't 
```
 implode("lpas[]=", $details_data['lpas'])
 ```
 Result in something more like
 ```
M-XYXY-YAGA-35G3lpas[]=M-XYXY-YAGA-35G2lpas[]=M-XYXY-YAGA-35G3
 ```
 ?
 
 I think you would need:
 ```
 implode("&lpas[]=", $details_data['lpas'])
 ```
 
 Note the extra `&`

Reply:
So 
```
"/start?personType=donor&lpas[]=" . implode("&lpas[]=", $details_data['lpas'])
```